core/template.js

```diff
@@ -27,28 +24,28 @@
 	 * @param {String} source The template source.
 	 */
 	CKEDITOR.template = function( source ) {
-		// Builds an optimized function body for the output() method, focused on performance.
 		// For example, if we have this "source":
 		//	'<div style="{style}">{editorName}</div>'
 		// ... the resulting function body will be (apart from the "buffer" handling):
 		//	return [ '<div style="', data['style'] == undefined ? '{style}' : data['style'], '">', data['editorName'] == undefined ? '{editorName}' : data['editorName'], '</div>' ].join('');
 
 		// Try to read from the cache.
-		if ( cache[ source ] )
+		if ( cache[ source ] ) {
 			this.output = cache[ source ];
+		}
 		else {
-			var fn = source
-				// Escape chars like slash "\" or single quote "'".
-				.replace( reEscapableChars, '\\$1' )
-				.replace( reNewLine, '\\n' )
-				.replace( reCarriageReturn, '\\r' )
-				// Inject the template keys replacement.
-				.replace( rePlaceholder, function( m, key ) {
-					return "',data['" + key + "']==undefined?'{" + key + "}':data['" + key + "'],'";
+			// less performant solution when CSP is disabling indirect evaluation
+			cache[ source ] = function( data, buffer ) {
```

Comment:
The previous implementation created a custom function for each template (something like compiling templates). This one also does this but the function is generic (basically, only `source` is different and the rest of the function is the same). With such approach we could just go with something like:

```
CKEDITOR.template = function( source ) {

	this.source = source;
	
};

CKEDITOR.template.prototype.output = function output( data, buffer ) {

	var output = '',
		source = this.source;

	output += String( source ).replace( rePlaceholder, function( fullMatch, dataKey ) {
		return data[ dataKey ] !== undefined ? data[ dataKey ] : fullMatch;
	} );

	return buffer ? buffer.push( output ) : output;
}
```

Shouldn't be much slower as the `new CKEDITOR.template( source );` is used anyway so the new object is created for each template.